File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    sentence
    |> String.downcase()
    |> String.split(~r/[^-a-z0-9\x7f-\xff']/, trim: true)
    |> Enum.map(&String.replace(&1, ~r/(^'|'$)/, ""))
    |> Enum.frequencies()
  end
end

File: python/word-count/word_count.py
import re
import string


def count_words(sentence):

    regex = "^'+|'+$|\\s'|'\\s|[^'a-zA-Z0-9]"

    words = re.sub(regex, " ", sentence.lower()).split()

    word_freq = {}

    for word in words:
        word_freq[word] = words.count(word)

    return word_freq
